src/Ocelot.Provider.Polly/PollyResiliencePipelineDelegatingHandler.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Ocelot.Configuration;
+using Ocelot.Logging;
+using Ocelot.Provider.Polly.Interfaces;
+using Polly.CircuitBreaker;
+using System.Diagnostics;
+
+namespace Ocelot.Provider.Polly;
+
+public class PollyResiliencePipelineDelegatingHandler : DelegatingHandler
+{
+    private readonly DownstreamRoute _route;
+    private readonly IHttpContextAccessor _contextAccessor;
+    private readonly IOcelotLogger _logger;
```

Comment:
_logger is not used, we should remove it

Reply:
OR... We could use the logger in the code 😉 👇 
```csharp
    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
    {
        var qoSProvider = this.GetQoSProvider();
        var pipeline = qoSProvider.GetResiliencePipeline(_route);

        if (pipeline == null)
        {
            _logger.LogDebug($"No {nameof(pipeline)} was detected by QoS provider for the route with downstream URL '{request.RequestUri}'.");
            return await base.SendAsync(request, cancellationToken); // shortcut > no qos
        }

        _logger.LogInformation($"A {nameof(pipeline)} has detected by QoS provider for the route with downstream URL '{request.RequestUri}'. Executing...");
        return await pipeline.ExecuteAsync(async (token) => await base.SendAsync(request, token), cancellationToken);
    }
```

Reply:
My opinion, this issue is major one. We shouldn't remove logger at all.

Reply:
Fixed in commit https://github.com/ThreeMammals/Ocelot/pull/1914/commits/9bcad647a07ce091c9f951e883964b35dec12baa ✔️